feat: use configured pk type in Action Mailbox migration

nshki opened the pull request:
<!--
Thanks for contributing to Rails!

Please do not make *Draft* pull requests, as they still send
notifications to everyone watching the Rails repo.

Create a pull request when it is ready for review and feedback
from the Rails team :).

If your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

About this template

The following template aims to help contributors write a good description for their pull requests.
We'd like you to provide a description of the changes in your pull request (i.e. bugs fixed or features added), motivation behind the changes, and complete the checklist below before opening a pull request.

Feel free to discard it if you need to (e.g. when you just fix a typo). -->

### Motivation / Background

<!--
Describe why this Pull Request needs to be merged. What bug have you fixed? What feature have you added? Why is it important?
If you are fixing a specific issue, include "Fixes #ISSUE" (replace with the issue number, remove the quotes) and the issue will be linked to this PR.
-->

I recently ran Action Mailbox's generators on a new Rails 7 project and noticed that it didn't carry over my generator primary key configuration. I had to manually modify the migration in my project. I know that Active Storage is a gem that uses a project's configured primary key, so I thought this would be a great opportunity for Action Mailbox to do the same.

### Detail

This PR adds a `migrations_test.rb` for Action Mailbox that closely mirrors the same file in Active Storage. It then adds a private `primary_key_type` method in the Action Mailbox migration which pulls a project's configured primary key type. This method is also closely modeled after Active Storage's.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

eileencodes commented on actionmailbox/db/migrate/20180917164000_create_action_mailbox_tables.rb:
```diff
@@ -10,4 +10,11 @@ def change
+    def primary_key_type
+      config = Rails.configuration.generators
+      setting = config.options[config.orm][:primary_key_type]
+      setting || :primary_key
```
I'd prefer to copy the exact code from ASt for this https://github.com/rails/rails/blob/main/activestorage/db/update_migrate/20191206030411_create_active_storage_variant_records.rb#L16-L19

eileencodes commented on actionmailbox/test/dummy/db/migrate/20180208205311_create_action_mailbox_tables.rb:
```diff
@@ -10,4 +10,11 @@ def change
+      config = Rails.configuration.generators
+      setting = config.options[config.orm][:primary_key_type]
+      setting || :primary_key
+    end
```
Same comment as the other `primary_key_type`.

eileencodes commented on actionmailbox/test/migrations_test.rb:
```diff
@@ -0,0 +1,54 @@
+    end
+
+    def primary_key(table)
+      puts "TEEHEE"
```
```suggestion
```

nshki replied:
Ha, this is embarrassing. Apologies for not taking out my debug comments!

eileencodes replied:
No worries, we've all done it 😄

eileencodes commented on actionmailbox/test/migrations_test.rb:
```diff
@@ -0,0 +1,54 @@
+
+    def primary_key(table)
+      puts "TEEHEE"
+      puts @connection.columns(table).inspect
```
```suggestion
```

eileencodes reviewed (changes requested):
Hi @nshki thanks for the PR! I've left some comment for changes that need to be made. Once you've made those changes please squash your commits into one. We generally don't merge multi-commit PRs that are related to a single change.

nshki committed:
feat: use config pk type in Action Mailbox migration

nshki commented:
Thanks for the review @eileencodes! I went ahead and addressed your comments and squashed the commits into one. 🙇🏻 I also renamed this PR title to be a little shorter so it's friendlier as a commit message.
